### PR TITLE
Enabling  ~~c++14~~ c++17

### PR DESCRIPTION
**Pull request by dcoeurjo**
# PR Description

Enabling ~~c++14~~ c++17 in DGtal

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug mode.
- [x] All continuous integration tests pass (Github Actions & appveyor)

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # DGtal 1.4beta
 
 ## New features / critical changes
+ 
+- *General*
+  - Major update: C++14 is now required for DGtal. (David Coeurjolly,
```
Comment (dcoeurjo) [inline ChangeLog.md:6]:
```suggestion
  - Major update: C++17 is now required for DGtal. (David Coeurjolly, 
```

File: cmake/Common.cmake [hunk 2]
```diff
@@ -30,7 +30,7 @@ endforeach()
 # -----------------------------------------------------------------------------
 # CPP11
 # -----------------------------------------------------------------------------
-set(DGTAL_CMAKE_CXX_STANDARD_MIN_REQUIRED 11)
+set(DGTAL_CMAKE_CXX_STANDARD_MIN_REQUIRED 14)
```
Comment (dcoeurjo) [inline cmake/Common.cmake:33]:
```suggestion
set(DGTAL_CMAKE_CXX_STANDARD_MIN_REQUIRED 17)
```

File: src/DGtal/doc/moduleBuildDGtal.dox [hunk 3]
```diff
@@ -33,7 +33,7 @@ manage the project build on various systems.
 To build DGtal in is minimal configuration, you need at least these
 dependencies:
 
-- \e c++ compiler (g++, clang++, ...) with C++11 features (gcc>4.6 for instance)
+- \e c++ compiler (g++, clang++, ...) with C++14 features
```
Comment (dcoeurjo) [inline src/DGtal/doc/moduleBuildDGtal.dox:36]:
```suggestion
- \e c++ compiler (g++, clang++, ...) with C++17 features
```

**Review by dcoeurjo — COMMENTED** [review]
Testing 17

**Review by kerautret — APPROVED** [review]
👍🏻 (I will update the DGtalTools)

**Comment by kerautret** [comment]
@dcoeurjo I tried some update to pass C++17 but still some issues due to QGLViewer, I tried to fix it but not success. I have put an issue [here](https://github.com/GillesDebunne/libQGLViewer/issues/67) when that should be fine to update I think..

**Comment by dcoeurjo** [comment]
Ok I see the issue on windows. Let's merge this PR anyway. LibQGLViewer bug is not really related to DGtal.

**Pull request merged**